### PR TITLE
Improve TimerTrigger error handling (#671)

### PR DESCRIPTION
**Pull request by mathewc**
Fix for https://github.com/Azure/azure-webjobs-sdk-extensions/issues/671

### CODE REVIEW

File: src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs [hunk 1]
```diff
@@ -174,25 +174,41 @@ public void Dispose()
             }
         }
 
-        private void OnTimer(object sender, ElapsedEventArgs e)
+        private async void OnTimer(object sender, ElapsedEventArgs e)
         {
-            HandleTimerEvent().Wait();
```
Comment (mathewc) [inline src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs:179]: Removing this Wait while I'm in here

File: src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs [hunk 2]
```diff
@@ -236,16 +252,12 @@ internal async Task InvokeJobFunction(DateTime invocationTime, bool isPastDue =
 
             try
             {
-                FunctionResult result = await _executor.TryExecuteAsync(input, token);
-                if (!result.Succeeded)
```
Comment (mathewc) [inline src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs:240]: Cleanup - no need to inspect then throw, if we're just swallowing the exception anyways

File: src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs [hunk 3]
```diff
@@ -174,25 +174,41 @@ public void Dispose()
...
+                    return;
+                }
 
-            StartTimer(DateTime.Now);
+                await InvokeJobFunction(DateTime.Now, false);
+            }
+            catch (Exception ex)
+            {
+                // ensure background exceptions don't stop the execution schedule
+                _logger.LogError(ex, "Error occurred during scheduled invocation.");
```
Comment (brettsam) [inline src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs:203]: Can you note the function name here? These can be hard to correlate if there's a bunch of timers in the app.

**Review by brettsam — APPROVED** [review]
Just one minor comment.

File: src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs (hunk 3 above)
Comment (mathewc) [reply src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs]: Done, thanks

**Pull request merged**